tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -80,6 +84,27 @@ public bool TryExtractGitMetadata([NotNullWhen(true)] out GitMetadata? gitMetada
         }
     }
 
+    private void PropagateGitMetadataToTheProfiler(GitMetadata gitMetadata)
+    {
+        try
+        {
+            // Avoid P/Invoke if the profiler is not ready (for obvious reason)
+            // but also if both repository url and commit sha are empty
+            if (Profiler.Instance.Status.IsProfilerReady &&
+                (!string.IsNullOrWhiteSpace(gitMetadata.RepositoryUrl) || !string.IsNullOrWhiteSpace(gitMetadata.CommitSha)))
```

Comment:
Nitpicking as usual so don't mind igoring, but wouldn't `gitMetadata != GitMetadata.Empty` be enough? Or have a `IsEmpty` helper in `GitMetadata` to have the logic there rather than here